### PR TITLE
Add timezone back in.

### PR DESCRIPTION
**Pull request by cdmurph32**
Update README with markdown generation step.

### CODE REVIEW

**Comment by pchickey** [comment]
Awesome! Can you also bump the clocks package version to `0.2.1-draft` as part of this PR?

**Comment by cdmurph32** [comment]
Sounds good.

Is there a way to update the version of wit-bindgen used for the CI job? The current version is v0.19.2.

**Comment by sunfishcode** [comment]
Updating to wit-abi-up-to-date@v19 in .github/workflows/main.yml should update CI to the latest wit-bindgen.

**Comment by cdmurph32** [comment]
If there aren't any other proposals for 0.2.1, I think the `-draft` should be removed before this is merged.
It needs to be removed before my wasmtime [PR](https://github.com/bytecodealliance/wasmtime/pull/6899/files).
Which in turn needs to be merged (among other things) before the iana-time-zone [PR](https://github.com/strawlab/iana-time-zone/compare/main...cdmurph32:iana-time-zone:wasi?expand=1).

**Comment by yoshuawuyts** [comment]
I've filed https://github.com/cdmurph32/wasi-clocks/pull/1 to add [`@unstable` feature gate](https://github.com/WebAssembly/component-model/pull/332) support to this PR. That should give us the ability to merge this into `wasi:clocks` as a feature which is independently tracked through the phases process. I've also filed https://github.com/WebAssembly/WASI/pull/605 to begin the process of tracking this extension, as well as reserve a canonical feature name (`clocks-timezone`).

**Comment by cdmurph32** [comment]
Thanks @yoshuawuyts . I've merged that.

File: wit/world.wit [hunk 1]
```diff
@@ -3,4 +3,5 @@ package wasi:clocks@0.2.0;
 world imports {
     import monotonic-clock;
     import wall-clock;
+    import timezone;
```
Comment (yoshuawuyts) [inline wit/world.wit:6]: ah shoot, I forgot to add the `@unstable` gate in my PR. We should change this to be:
```suggestion
    @unstable(feature = clocks-timezone)
    import timezone;
```

**Comment by yoshuawuyts** [comment]
@cdmurph32 oh heh, one more thing: there is still a merge conflict on this PR. Could you please rebase it on the main branch? I'm hoping that's the last bit left to get this to a state where it can be merged.

**Comment by cdmurph32** [comment]
Do we need to update wit-bindgen?
```
wit-bindgen markdown wit --html-in-md
Error: failed to resolve directory while parsing WIT for path [wit]

Caused by:
    0: this type is not gated by a feature but its interface is gated by a feature
    1: found a reference to a interface which is excluded due to its feature not being activated
            --> wit/timezone.wit:5:21
             |
           5 |     use wall-clock.{datetime};
             |                     ^-------
```

**Comment by yoshuawuyts** [comment]
Oh right, yeah — `wit-bindgen` has not been updated yet with support for the new gates notation. We probably need to cut a release of `wasm-tools` for that first too.

**Comment by sunfishcode** [comment]
wasm-tools and wit-bindgen are released, and I've now bumped their versions in wit-abi-up-to-date v20. So now, it should be sufficient to just bump the wit-abi-up-to-date version to v20 in .github/workflows/main.yml.

**Comment by cdmurph32** [comment]
Looks like the version of wit-bindgen-cli in the test needs to be updated.

**Comment by yoshuawuyts** [comment]
I've filed https://github.com/WebAssembly/wasi-clocks/pull/67 to update CI, which brings in https://github.com/WebAssembly/wit-abi-up-to-date/pull/26 which was published as part of [WebAssembly/wit-abi-up-to-date@v20](https://github.com/WebAssembly/wit-abi-up-to-date/releases/tag/v20). I think if we merge that, and then re-run CI here, we should be good.

**Comment by sunfishcode** [comment]
#67 is now merged, so now this should just need a rebase.

**Comment by cdmurph32** [comment]
CI won't pass until we can pass features to wit-abi-up-to-date.
```
 wit-bindgen markdown wit --check --html-in-md --features clocks-timezone
```

**Comment by sunfishcode** [comment]
With wit-abi-up-to-date v21, it should be possible to pass features to the script.

**Comment by cdmurph32** [comment]
Thank you @sunfishcode and @yoshuawuyts !

**Comment by justingrant** [comment]
Hi - I'm one of the champions of the ECMAScript [Temporal](https://github.com/tc39/proposal-temporal) proposal that will add a richer and timezone-aware set of date/time types to JS engines. @littledan suggested that I take a look at the WASI timezone proposal.

I had a few questions: 

* Is `name` intended to be a localized string suitable for end users, like "PDT"?  Or a software-friendly identifier from the IANA Time Zone Database? Or something else?  I'd strongly suggest that the latter is what platforms should offer, because strings like "PDT" are not standardized so it's hard to use them in software, other than for displaying to end users. And if you're intending to use them to display to end users, then those strings should be localized.
* Is there a reason you chose seconds as opposed to a more fine-grained unit for the time zone offset? I'm asking because there are historical time zones with sub-second offsets, and because in ECMAScript we use nanoseconds to represent time zone offsets.
* In other contexts, a simple "is DST" boolean has had some challenges when applied to some corner cases, e.g. https://github.com/moment/moment-timezone/issues/968.  What are the use cases that you're looking to support with this boolean?
* What kind of interaction (if any) would you expect to have with Temporal and with ECMAScript overall? Should WASI and ECMAScript date/time types be round-trippable?

cc @ptomato @sffc

**Comment by justingrant** [comment]
@cdmurph32 - I think you'll want @sffc to weigh in here.  He's both a Rust expert and a localization guru, and both may be involved in your question above.

Also, is the reason you're not using ICU4X (the Rust implementation of ICU4C/ICU4J) that WASI is intentionally minimal in size and you don't need the full capabilities of ICU? I assume that ICU4X does everything that iana-time-zone does, and much more. @sffc is one of the folks driving ICU4X.

Also, is the system time zone the only time zone that WASI is worried about?  There's no need to be able to calculate the offset for a different time zone?

Finally, where are you getting your IANA Time Zone Database data from in order to calculate the offset for the zone at that instant?

**Comment by yoshuawuyts** [comment]
Thanks for the feedback @justingrant! - I think it's clear we have room to improve, and we should figure out what the right steps forward are. To that effect I've filed https://github.com/WebAssembly/wasi-clocks/issues/69 to continue our conversation about how we can fix the issues in the current design. Let me know if anything is missing there!

**Pull request merged**